app/services/ExportedPlasticAnswer.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import controllers.helpers.NonExportedAmountHelper
+import models.UserAnswers
+import pages.returns._
+
+import scala.util.Try
+
+// Todo: This may be rename better.
+class ExportedPlasticAnswer(userAnswers: UserAnswers) {
+
+  def resetExportedByYouIfAllExportedPlastic: Try[UserAnswers] = {
+    if(isAllPlasticExported) {
+      userAnswers.set(DirectlyExportedComponentsPage, true, cleanup = false).get
+        .set(PlasticExportedByAnotherBusinessPage, false).get
+        .set(AnotherBusinessExportWeightPage, 0L, cleanup = false).get
+        .set(NonExportedHumanMedicinesPlasticPackagingPage, false, cleanup = false).get
+        .set(NonExportedHumanMedicinesPlasticPackagingWeightPage, 0L, cleanup = false).get
+        .set(NonExportedRecycledPlasticPackagingPage, false, cleanup = false).get
+        .set(NonExportedRecycledPlasticPackagingWeightPage, 0L, cleanup = false)
+    }
+    else {
+      userAnswers.set(DirectlyExportedComponentsPage, true, cleanup = false)
+    }
+  }
+
+  def resetAnotherBusinessIfAllExportedPlastic: Try[UserAnswers] = {
+    if(isAllPlasticExported) {
+      userAnswers
+        .set(DirectlyExportedComponentsPage, true, cleanup = false).get
+        .set(NonExportedHumanMedicinesPlasticPackagingPage, false, cleanup = false).get
+        .set(NonExportedHumanMedicinesPlasticPackagingWeightPage, 0L, cleanup = false).get
+        .set(NonExportedRecycledPlasticPackagingPage, false, cleanup = false).get
+        .set(NonExportedRecycledPlasticPackagingWeightPage, 0L, cleanup = false)
+    }
+    else {
+      userAnswers.set(PlasticExportedByAnotherBusinessPage, true, cleanup = false)
+    }
+  }
+
+  def resetAllIfNoTotalPlastic: UserAnswers = {
+
+    if(NonExportedAmountHelper.totalPlastic(userAnswers).getOrElse(0L) > 0L) {
+      userAnswers.copy()
```

Comment:
this does nothing?

Reply:
yep agree

Reply:
but just wanted to return a new ref instead of the same one so to keep with the concept of functional programming that a function should always return a new object. What do you think of that? Should it return just the same userAnswer?